### PR TITLE
Fix torch.view and close #9665

### LINKED ISSUE #9665: frontend for `torch.Tensor.view` (incomplete behaviour)
The `view` method of `torch.Tensor` allows the use of both a tuple and various `int`s as arguments. The frontend function does not support the expanded alternative as shown below. On top of that, it seems like the keyword argument is `size` and not `shape`.

```
import torch

x = torch.arange(8)
x_torch = x.view(size=(2, 2, 2))  # works
x_torch = x.view(shape=(2, 2, 2))  # fails
```

```
import torch
import ivy.functional.frontends.torch as torch_frontend
import ivy

ivy.set_backend("torch")  # "numpy", "jax", "tensorflow"

# Torch
x = torch.arange(8)
x_torch = x.view(2, 2, 2)  # works

# Torch frontend
x = torch_frontend.arange(8)
x_ivy = x.view(2, 2, 2)  # fails
```

### PR DESCRIPTION
**Pull request by mattiagaggi**
This closes #9665 issue.
Issues:
- keyword shape in torch.view used instead of size
- frontend of view failing

Changes made
- now the view function can take in integers, a tuple of integers or the argument size 
- added a test to verify view works as expected
- removed * from view tensor as not necessary

Further comments:
The confusion shape/size comes from the pytorch documentation (they actually do say the keyword is shape which is wrong) and the fact that the current implementation of view is in C. I found the old one here (https://github.com/pytorch/pytorch/blob/518864a7e094f32044ef4c0de82c0f20f4ed99c2/torch/tensor.py#L178-L217) which is not great.

I am not super happy about this new implementation, but it works. I think the idea of being as restrictive as we can with the inputs to view could help.

### CODE REVIEW

File: ivy/functional/frontends/torch/tensor.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # local
+import weakref
```
[the remark this answers is not in the document]
Comment (mattiagaggi) [reply ivy/functional/frontends/torch/tensor.py:2]: I just used black and isort to format, they changed that automatically

File: ivy/functional/frontends/torch/view_tensor.py [hunk 2]
```diff
@@ -47,14 +48,14 @@ def new_method(self, *args, **kwargs):
 
 
 class ViewTensor:
-    def __init__(self, ref, *, shape):
+    def __init__(self, ref, size):
```
[the remark this answers is not in the document]
Comment (mattiagaggi) [reply ivy/functional/frontends/torch/view_tensor.py:51]: ah yeah you are right good spot!

**Comment by mattiagaggi** [comment]
> it seems like your change make some other functions using tensor fails? ivy/functional/backends/torch/manipulation.py:110: in reshape newarr = torch.clone(x) E TypeError: clone(): argument 'input' (position 1) must be Tensor, not numpy.ndarray

I think what could cause an error in the backend is that my function doesn't support list of ints as inputs. I have just changed that so we'll see.

**Comment by mattiagaggi** [comment]
> > it seems like your change make some other functions using tensor fails? ivy/functional/backends/torch/manipulation.py:110: in reshape newarr = torch.clone(x) E TypeError: clone(): argument 'input' (position 1) must be Tensor, not numpy.ndarray
> 
> I think what could cause an error in the backend is that my function doesn't support list of ints as inputs. I have just changed that so we'll see.

After changing that I rechecked all the usages of the view function and couln't find anything else. Possible that fixing .view has revealed more issues? or maybe some other PR?

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
well, the same kind of error is still there:
ivy/functional/backends/torch/manipulation.py:[110](https://github.com/unifyai/ivy/actions/runs/3933059752/jobs/6726353188#step:3:111): in reshape
    newarr = torch.clone(x)
E   TypeError: clone(): argument 'input' (position 1) must be Tensor, not numpy.ndarray

**Comment by mattiagaggi** [comment]
> well, the same kind of error is still there: ivy/functional/backends/torch/manipulation.py:[110](https://github.com/unifyai/ivy/actions/runs/3933059752/jobs/6726353188#step:3:111): in reshape newarr = torch.clone(x) E TypeError: clone(): argument 'input' (position 1) must be Tensor, not numpy.ndarray

I reverted the commits, now aside from different spacing, everything is exactly the same as the master branch. The test still fail so it is not my code that failed

**Review by Daniel4078 — COMMENTED** [review]
Now I see, then I suppose your origenal implementation is not wrong.

**Comment by mattiagaggi** [comment]
> Now I see, then I suppose your origenal implementation is not wrong.

what should I do then? kind of need to get this in to progress to the interview.

**Review by Daniel4078 — COMMENTED** [review]
I suppose you can just add your changes back and then I can merge it for you.

**Comment by mattiagaggi** [comment]
> I suppose you can just add your changes back and then I can merge it for you.

Just reverted to previous commit. all good to merge

**Review by Daniel4078 — APPROVED** [review]
can't find anything wrong, well done.

**Comment by mattiagaggi** [comment]
> can't find anything wrong, well done.

thanks mate! would you mind pinging the team to move forward for the next steps :). Really great working with you what is your discord handle btw?

**Comment by Daniel4078** [comment]
Well, I have no control over the hiring progress at all, and I can only hope you good luck after you submit the link of this pull request.

**Comment by mattiagaggi** [comment]
> Well, I have no control over the hiring progress at all, and I can only hope you good luck after you submit the link of this pull request.

of course man, I just assume I have to let somebody know my PR is done right? otherwise how would they know

**Comment by Daniel4078** [comment]
My experience about this is you just need to submit the link to this pull request through a link you get from the email and then wait for the next email to come.

**Comment by mattiagaggi** [comment]
> My experience about this is you just need to submit the link to this pull request through a link you get from the email and then wait for the next email to come.

got it, yes I have it. Thank you so much!

**Pull request merged**